### PR TITLE
MWP consumer app for testing

### PR DESCRIPTION
**Pull request by mmcgahan**
New package, but this PR is mostly for sharing investigation progress for now

### CODE REVIEW

**Comment by mmcgahan** [comment]
baseline metrics from `mwp-app-base` branch that tests the version of the platform that is running in mup-web:

To get the most reliable numbers, I think you have to start the server and let it sit for about a minute and wait for _4_ garbage collections happen. Once that happens, the server appears to be in a 'steady state' and can be tested with `autocannon -H Cookie=$COOKIE -c 20 -a 20000 https://beta2.dev.meetup.com:8000/`

Interestingly, the rate of increase in memory usage drops off logarithmically

```
...
Memory leak detected:
  reason: 'heap growth over 5 consecutive GCs (4s) - 1386.15 mb/hr'
...
Memory leak detected:
  reason: 'heap growth over 5 consecutive GCs (7s) - 494.95 mb/hr'
...
Memory leak detected:
  reason: 'heap growth over 5 consecutive GCs (7s) - 262.21 mb/hr'
...
```

File: packages/mwp-api-state/src/cache/index.test.js [hunk 1]
```diff
@@ -25,31 +21,27 @@ function makeCacheEpic() {
...
-	const apiSuccessAction$ = ActionsObservable.of(MOCK_SUCCESS_ACTION);
-	return CacheEpic(apiSuccessAction$).toPromise().then(() => CacheEpic);
+	return CacheEpic(MOCK_SUCCESS_ACTION).then(() => CacheEpic);
 }
 
 function clearCacheEpic(CacheEpic) {
 	// clear the cache with CACHE_CLEAR
-	const clearAction$ = ActionsObservable.of({ type: 'CACHE_CLEAR' });
-	return CacheEpic(clearAction$).toPromise().then(() => CacheEpic);
+	return CacheEpic({ type: 'CACHE_CLEAR' }).then(() => CacheEpic);
```
Comment (eilinora) [inline packages/mwp-api-state/src/cache/index.test.js]: suggest using the `cache_clear` const for consistency

File: packages/mwp-api-state/src/cache/index.test.js [hunk 2]
```diff
@@ -25,31 +21,27 @@ function makeCacheEpic() {
...
 
-const testForPopulatedCache = (action = apiRequestAction) => CacheEpic => {
-	const testAction$ = ActionsObservable.of(action);
-	return CacheEpic(testAction$)
-		.do(action => expect(action.type).toEqual('CACHE_SUCCESS'))
-		.toPromise();
-};
+const testForPopulatedCache = (action = apiRequestAction) => CacheEpic =>
+	CacheEpic(action).then(actions =>
+		expect(actions.map(({ type }) => type)).toContain('CACHE_SUCCESS')
```
Comment (eilinora) [inline packages/mwp-api-state/src/cache/index.test.js]: same about using const here

File: packages/mwp-api-state/src/sync/index.js [hunk 3]
```diff
@@ -64,85 +52,74 @@ export function getDeprecatedSuccessPayload(successes, errors) {
...
+		const state = store.getState();
+		const { referrer = {} } = state.routing;
+		// inject request metadata from context, including `store.getState()`
+		const requestMetadata = {
+			referrer: referrer.pathname || state.config.entryPath || '',
+			logout: location.pathname.endsWith('logout'), // assume logout route ends with logout
+			clickTracking: state.clickTracking,
+			retainRefs: [],
+		};
+		const cacheAction = requestMetadata.logout && { type: 'CACHE_CLEAR' };
```
Comment (eilinora) [inline packages/mwp-api-state/src/sync/index.js]: same about const

**Pull request merged**